Support php 8.2

aashan10 opened the pull request:
<!--

Before working on a contribution, you must determine on which branch you need to work:
- Bug fix: choose the latest maintenance branch `10.5`
- Feature/Improvement: choose `11.x` 

> All bug fixes merged into the latest maintenance branch are also merged to the current dev branch (`11.x`) on a regular basis.

## Please make sure your PR complies with all of the following points: 
- [ ] Read and accept our [contributing guidelines](/CONTRIBUTING.md) before you submit a PR.
- [ ] Features need to be proper documented in `doc/` 
- [ ] Bugfixes need a short guide how to reproduce them -> target branch is the oldest supported maintenance branch, e.g. `10.5` (see Readme.md for the list of supported versions)
- [ ] Meet all coding standards (see PhpStan actions) 

**Don't submit a PR if it doesn't comply, it'll be closed without a comment!**
-->  
  

## Changes in this pull request  
Resolves #

## Additional info

brusch committed:
[Actions] PhpStan workflow: added PHP 8.2

brusch committed:
[Actions] Codeception workflow: added PHP 8.2

aashan10 committed:
Fixed phpstan issues

aashan10 committed:
Updated definition of Dataobject

aashan10 committed:
Fixed phpstan errors

aashan10 committed:
Fixed phpstan issues

aashan10 committed:
Fixed phpstan issues

aashan10 committed:
Merge branch '11.x' into php_8_2

aashan10 committed:
Merge branch '11.x' of https://github.com/pimcore/pimcore into php_8_2

aashan10 committed:
Merge branch 'php_8_2' of https://github.com/aashan10/pimcore into php_8_2

aashan10 committed:
Merge branch '11.x' of https://github.com/pimcore/pimcore into php_8_2

aashan10 committed:
Fixed phpstan issue

aashan10 committed:
Updated carbon version to latest

aashan10 commented:
@dvesh3 Please use this PR in favor of https://github.com/pimcore/pimcore/pull/13861 as @brusch is not available and I do not have permission to push commits into the existing pr

blankse commented on bundles/EcommerceFrameworkBundle/src/CoreExtensions/ClassDefinition/IndexFieldSelectionCombo.php:
```diff
@@ -47,27 +47,25 @@ protected function buildOptions(): array
-                }
-            } catch (\Exception $e) {
-                Logger::error((string) $e);
+        try {
```
We should revert this and add it to the baseline file again.

aashan10 committed:
Revert phpstan issue

kingjia90 commented:
LGTM, tests are green as well.

It seems it might depend on https://github.com/pimcore/pimcore/pull/13888, according to this [comment ](https://github.com/pimcore/pimcore/pull/13861#issuecomment-1363247376), just wondering if there's any conflicting/duplicate changes between these 2 PRs, @aashan10 could you please verify this and if we are still depending on that?

kingjia90 commented on models/DataObject/ClassDefinition/Data/StructuredTable.php:
```diff
@@ -66,6 +66,20 @@ class StructuredTable extends Data implements ResourcePersistenceAwareInterface,
+     *
+     * @var string|array|null
+     */
+    protected $columnType = null;
```
Just quickly checked, these additions (eg. [columnType = null](https://github.com/pimcore/pimcore/pull/13888/files/bb5fed108c73628521e9c8267fdcb33483b66bac?w=1#diff-88c43c7b1020461fcf7abcd0a6e7f4ecd4f0d71466b0944f71ce762877647f33)) are going to be osbolete and removed.

The rest looks good, not sure if this PR could be merged first and delegate the full clean up in  https://github.com/pimcore/pimcore/pull/13888?

kingjia90 commented:
TODO:
- [x] Add an upgrade note about the support of 8.2 in the 11 paragraph.
- [x] ~Re-adapt the changes (composer, system requirement doc) on https://github.com/pimcore/pimcore/pull/14180 after release (when 10.5,10.6 get merged into 11.x)~ solved in merge conflict https://github.com/pimcore/pimcore/commit/e3c005abf4760d96f32bf5f743aa7fc4f6290cfd

kingjia90 committed:
Merge remote-tracking branch 'origin/11.x' into php_8_2

kingjia90 committed:
Task: add changelog entry

jdreesen commented on doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:
```diff
@@ -2,8 +2,9 @@
 ## 11.0.0
 - [General] **Attention:** Added native php types for argument types, property types, return types and strict type declaration where possible. 
  This results in **various bc breaks**. Please make sure to add the corresponding types to your implementation.
+- [Symfony] Bumped Symfony packages to "^6.1". Pimcore 11 will only support Symfony 6.
```
Note that [Symfony 6.1 is already EOL](https://symfony.com/releases/6.1). Maybe bump directly to 6.2?

kingjia90 replied:
That's a good point, didn't even realized that 6.1 being EOL and just 3 days ago 😮 Thak you for the heads-up!

We are bumping to 6.2 in the [PR removing SensioFrameworkExtraBundle ](https://github.com/pimcore/pimcore/pull/14062/files#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34), so i will add the doc change over there before merging it

jdreesen commented on doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md:
```diff
@@ -2,8 +2,9 @@
 - [General] **Attention:** Added native php types for argument types, property types, return types and strict type declaration where possible. 
  This results in **various bc breaks**. Please make sure to add the corresponding types to your implementation.
+- [Symfony] Bumped Symfony packages to "^6.1". Pimcore 11 will only support Symfony 6.
+- [PHP] The minimum supported PHP version is now 8.1 and added support for 8.2 
```
I think you should change the PHP requirement in `composer.json` from `^8.1` to `~8.1.0 || ~8.2.0` if you don't plan (or don't know yet) to support PHP 8.3.

kingjia90 replied:
Since PHP 8.3 is still in work-in-progress, nobody knows if it will be supported, but it would be safe to assume it will be supported by Pimcore (like a "until proved otherwise" 😄) and that we could later restrict maximum to 8.2 in a bug fix release like https://github.com/pimcore/pimcore/pull/14180 if turns out differently

jdreesen replied:
As safe as it was to assume that Pimcore 10 would support PHP 8.2? ;)

The problem is: if you have a release (let's say `v11.0`) that states it supports PHP 8.3, composer will install it. If you later disallow this PHP version in say `v11.1` and someone tries to install Pimcore 11 via `composer require pimcore/pimcore` while running on PHP 8.3, composer will install `v11.0` (because `11.1` doesn't support the PHP version, but `v11.0` does (or at least says it does)). Now, if we are in a situation like right now, where you decided to support PHP 8.3 only with the next mayor version Pimcore 12, but it is not released yet, people will get a "broken" version, and maybe it runs, but only a few features are broken that aren't used very often, so they won't notice immediately. That's a big source of frustration.

brusch replied:
@jdreesen thanks for the input, you're right, `~8.1.0 || ~8.2.0` probably makes more sense. I'd go for it as it is more safe. Thanks!

kingjia90 commented on models/DataObject/Concrete.php:
```diff
@@ -758,7 +765,7 @@ public function __wakeup(): void
         // renew localized fields
         // do not use the getter ($this->getLocalizedfields()) as it somehow slows down the process around a sec
         // no clue why this happens
-        if (property_exists($this, 'localizedfields') && $this->localizedfields instanceof Localizedfield) {
```
The latest test fail errors and this change may solved by #13874

kingjia90 committed:
Chore: set maximum php 8.2

mcop1 committed:
Merge remote-tracking branch 'origin/11.x' into php_8_2

# Conflicts:
#	.github/workflows/codeception.yaml
#	.github/workflows/static-analysis.yaml
#	doc/Development_Documentation/23_Installation_and_Upgrade/09_Upgrade_Notes/README.md
#	models/DataObject/ClassDefinition/Data/Extension/QueryColumnType.php
#	models/DataObject/Concrete.php

mcop1 committed:
Removed unnecessary properties

mcop1 committed:
Removed $renderer property, directly return string instead
Set php 8.2 tests from experimental true to false and added tests for symfony 6 dev-branch and php 8.2

mcop1 committed:
Removed symfony dev branch tests for 8.1, dev branch tests with 8.2 are enough

mcop1 committed:
Removed upgrade notes entry, it was already added by another pr
